prometheus/src/main/java/org/opensearch/sql/prometheus/storage/PrometheusStorageEngine.java

```diff
@@ -7,34 +7,58 @@
 
 package org.opensearch.sql.prometheus.storage;
 
+import static org.opensearch.sql.utils.SystemIndexUtils.isSystemIndex;
+
 import java.util.Collection;
 import java.util.Collections;
+import lombok.RequiredArgsConstructor;
+import org.opensearch.sql.CatalogSchemaName;
+import org.opensearch.sql.analysis.model.SchemaName;
+import org.opensearch.sql.exception.SemanticCheckException;
 import org.opensearch.sql.expression.function.FunctionResolver;
 import org.opensearch.sql.prometheus.client.PrometheusClient;
 import org.opensearch.sql.prometheus.functions.resolver.QueryRangeTableFunctionResolver;
+import org.opensearch.sql.prometheus.storage.system.PrometheusSystemTable;
 import org.opensearch.sql.storage.StorageEngine;
 import org.opensearch.sql.storage.Table;
+import org.opensearch.sql.utils.SystemIndexUtils;
 
 
 /**
  * Prometheus storage engine implementation.
  */
+@RequiredArgsConstructor
 public class PrometheusStorageEngine implements StorageEngine {
 
   private final PrometheusClient prometheusClient;
 
-  public PrometheusStorageEngine(PrometheusClient prometheusClient) {
-    this.prometheusClient = prometheusClient;
+  @Override
+  public Collection<FunctionResolver> getFunctions() {
+    return Collections.singletonList(
+        new QueryRangeTableFunctionResolver(prometheusClient));
   }
 
   @Override
-  public Table getTable(String name) {
-    return null;
+  public Table getTable(CatalogSchemaName catalogSchemaName, String tableName) {
+    if (isSystemIndex(tableName)) {
+      return new PrometheusSystemTable(prometheusClient, catalogSchemaName, tableName);
+    } else if (SchemaName.INFORMATION_SCHEMA_NAME.equals(catalogSchemaName.getSchemaName())) {
+      return resolveInformationSchemaTable(catalogSchemaName, tableName);
+    } else {
+      return null;
```

Comment:
Could you avoid returnning null pointer?

Reply:
In the next PR, when we support PPL commands. This won't be null. This is temporary.

Reply:
please add the Todo as comments.

Reply:
ok